triton/scripts/onnx_triton_utils.py

```diff
@@ -126,7 +126,16 @@ def forward(
             2  # we will cut off 1 frame on each side of encoder_embed output
         )
 
-        src_key_padding_mask = make_pad_mask(lengths)
+        embed = self.encoder_embed(x)
+
+        # cut off 1 frame on each size of embed as they see the padding
+        # value which causes a training and decoding mismatch.
+        embed = embed[:, 1:-1, :]
+
+        embed, pos_enc = self.encoder_pos(embed, self.left_context)
+        embed = embed.permute(1, 0, 2)  # (B, T, F) -> (T, B, F)
+
+        src_key_padding_mask = make_pad_mask(lengths, embed.size(0))
```

Comment:
(Btw, this is where I have to make this PR https://github.com/k2-fsa/icefall/pull/670 for dynamic batch with paded audios. )